File: 0009-palindrome-number.c
//翻转比较是否相同
bool isPalindrome(int x) {
    if(x < 0) {
        return false;
    }
    long temp = x, y = 0;
    while(temp) {
        y = y * 10 + temp % 10;
        temp /= 10;
    }
    return x == y;
}

File: 0009-palindrome-number/0009-palindrome-number.py
# 反转数之后比较
class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x < 0:
            return False
        if x == 0:
            return True
        x1 = x
        num = 0
        while x > 0:
            num = num * 10 + (x % 10)
            x = x // 10
        return x1 == num
